addon/commands/insert-table-command.ts

```diff
@@ -0,0 +1,109 @@
+import Command from "./command";
+import Model from "@lblod/ember-rdfa-editor/model/model";
+import ModelSelection from "@lblod/ember-rdfa-editor/model/model-selection";
+import ModelTable from "@lblod/ember-rdfa-editor/model/model-table";
+import ModelNode from "../model/model-node";
+import {MisbehavedSelectionError, NoParentError, SelectionError} from "@lblod/ember-rdfa-editor/utils/errors";
+import {INVISIBLE_SPACE} from "../model/util/constants";
+
+
+export default class InsertTableCommand extends Command {
+  name = "insert-table";
+
+  constructor(model: Model) {
+    super(model);
+  }
+
+  canExecute(): boolean {
+    return true;
+  }
+
+  execute(): void {
+
+    const selection= this.model.selection;
+    if (!ModelSelection.isWellBehaved(selection)) {
+      throw new MisbehavedSelectionError();
+    }
+
+    const selectionStart = selection.lastRange.start;
+    const selectionEnd = selection.lastRange.end;
+    const startPos = selectionStart.parentOffset;
+    const endPos = selectionEnd.parentOffset;
+
+    const selectedIterator = selection.findAllInSelection({});
+
+    if (!selectedIterator) {
+      // should be impossible
+      throw new SelectionError("couldn't get iterator");
+    }
+
+    const selected = Array.from(selectedIterator);
+
+    // get the first and last textnodes of the selection
+    const firstText = selected.find(ModelNode.isModelText);
+    const lastText = selected.reverse().find(ModelNode.isModelText);
+
+    if (!firstText || !lastText) {
+      throw new SelectionError("No text nodes in selection");
+    }
+
+    // split the node at the start of the selection
+    const {left: leftOfStart, right: rightOfStart} = firstText.split(startPos);
+    // split the node at the end of the selection
+    const {left: leftOfEnd, right: rightOfEnd} = lastText.split(endPos);
+
+    const leftParent = leftOfStart.parent;
+    if (!leftParent) {
+      throw new NoParentError();
+    }
+
+    const table = new ModelTable(2,2);
+
+    //handle zero length selection
+    if (selection.isCollapsed) {
+      // add the break to the right of the node before the cursor
```

Comment:
try to keep the comments up to date when copying code

Reply:
What's wrong about those commands? I keep them because I didn't change any code and I think they describe good what the code is doing